packages/jaeger-ui/src/components/TracePage/TraceStatistics/generateDropdownValue.tsx

```diff
@@ -41,11 +45,9 @@ function getValueTagIsPicked(tableValue: ITableSpan[], trace: Trace, nameSelecto
   }
   availableTags = [...new Set(availableTags)];
 
-  const tags = _(availableTags).map('tags').flatten().value();
-  let tagKeys = _(tags).map('key').uniq().value();
-  tagKeys = _.filter(tagKeys, function calc(o) {
-    return o !== nameSelectorTitle;
-  });
+  const tags = _map(availableTags, 'tags');
+  let tagKeys = _uniq(_map(_flatten(tags), 'key'));
+  tagKeys = _filter(tagKeys, o => o !== nameSelectorTitle);
```

Comment:
This may be the equivalent, but could we do better? First, availableTags var is overloaded: starts a set of spans, then set of strings, bad design. And all these lodash ops are revisiting the same data that was already processed in the loop above - could we just calculate what we need right there?

Reply:
At least we should rename the var to spansWithFilterTag

Reply:
I tried playing with the former logic but it is getting a bit complex to accommodate the work we are taking out of lodash functions to the loops.